nav2_smac_planner/src/a_star.cpp

```diff
@@ -229,15 +290,17 @@ bool AStarAlgorithm<NodeT>::areInputsValid()
   }
 
   // Check if points were filled in
-  if (!_start || !_goal) {
+  if (!_start || _goalsSet.empty()) {
     throw std::runtime_error("Failed to compute path, no valid start or goal given.");
   }
 
   // Check if ending point is valid
-  if (getToleranceHeuristic() < 0.001 &&
-    !_goal->isNodeValid(_traverse_unknown, _collision_checker))
-  {
-    throw nav2_core::GoalOccupied("Goal was in lethal cost");
+  if (getToleranceHeuristic() < 0.001) {
+    // if any goal is valid, then all goals are valid
+    auto goal = *_goalsSet.begin();
```

Comment:
Q: should we be pruning goals that are invalid if any are valid? (i.e. check if any not in collision, if so, prune any that are in collision. If not prune none since they're all going to be approximate)